### PR TITLE
VSCode: Create `Context` class

### PR DESCRIPTION
**Pull request by mkaput**
**Stack**:
- #5015
- #5014
- #5013
- #5012
- #5011
- #4976
- #4975
- #4974
- #4973
- #4959
- #4935
- #4934 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4934)

### CODE REVIEW

File: vscode-cairo/src/extension.ts [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-export async function activate(context: vscode.ExtensionContext) {
-  const config = vscode.workspace.getConfiguration();
-  const outputChannel = vscode.window.createOutputChannel("Cairo extension");
-  context.subscriptions.push(outputChannel);
+export async function activate(extensionContext: vscode.ExtensionContext) {
+  const ctx = Context.create(extensionContext);
 
-  if (config.get<boolean>("cairo1.enableLanguageServer")) {
-    client = await setupLanguageServer(config, context, outputChannel);
+  if (ctx.config.get<boolean>("enableLanguageServer")) {
```
Comment (Arcticae) [inline vscode-cairo/src/extension.ts:11]: It'd be nice to keep the configs somewhere as consts and behind some typed facade

**Review by Arcticae — APPROVED** [review]

File: vscode-cairo/src/extension.ts [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+import { Context } from "./context";
 
 let client: lc.LanguageClient | undefined;
 
-export async function activate(context: vscode.ExtensionContext) {
-  const config = vscode.workspace.getConfiguration();
-  const outputChannel = vscode.window.createOutputChannel("Cairo extension");
-  context.subscriptions.push(outputChannel);
+export async function activate(extensionContext: vscode.ExtensionContext) {
+  const ctx = Context.create(extensionContext);
```
Comment (Arcticae) [inline vscode-cairo/src/extension.ts:9]: I wonder if that isn't something that we'd like to keep around globally for easier import and avoidance of prop-drilling

**Review by Arcticae — COMMENTED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4934#-:-Nq7GjZW3Fu7WyY2YxVv:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @orizi)

**Comment by mkaput** [comment]
Closing pull request: commit has gone away

**Review by orizi — APPROVED** [review]
:lgtm:

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/4934#-:-NqPCBzjEXYQR9yotoGb:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @mkaput)

**Pull request merged**